src/lib.rs

```diff
@@ -350,9 +353,9 @@ impl<'a> Write for &'a NamedPipe {
         }
 
         // Move `buf` onto the heap and fire off the write
-        //
-        // TODO: need to be smarter about buffer management here
-        Inner::schedule_write(&self.inner, buf.to_vec(), 0, &mut io);
+        let mut owned_buf = self.inner.get_buffer();
+        owned_buf.extend(buf);
```

Comment:
It seems `extend` will call `heap::reallocate` when there is no enough memory. Does it need to pre-allocate some space for the buffer we got from `pool`? It seems the `buf` would need to be reallocated when *Audio-IPC* send a super big message suddenly. Or this won't happen?

Reply:
That's expected.  Whenever the pool is exhausted or an allocation larger than the pool can provide is requested, we just fall back to the old behaviour of allocating as necessary.

Almost all AudioIPC messages fit into small buffers, so once the pipe is running there are ~0 allocations for the common case.  There a few larger AudioIPC messages (e.g. enumerated device array), but those aren't a concern as they are fairly infrequent.